lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -577,464 +665,528 @@
-            return supportedLocalesOfWrapper(this, numberFormat_supportedLocalesOf_name, locales, options);
-        }), intlStaticMethodThisArg);
+    const numberFormat_supportedLocalesOf_func = tagPublicFunction(numberFormat_supportedLocalesOf_name, function (locales, options) {
+        return supportedLocalesOf_unconstructable(this, numberFormat_supportedLocalesOf_name, undefined, locales, options);
+    });
+    const numberFormat_supportedLocalesOf = _.bind(numberFormat_supportedLocalesOf_func, intlStaticMethodThisArg);
 
     const dateTimeFormat_supportedLocalesOf_name = "Intl.DateTimeFormat.supportedLocalesOf";
-    const dateTimeFormat_supportedLocalesOf = callInstanceFunc(FunctionInstanceBind, tagPublicFunction(dateTimeFormat_supportedLocalesOf_name,
-        function dateTimeFormat_supportedLocalesOf_dummyName(locales, options = undefined) {
-            return supportedLocalesOfWrapper(this, dateTimeFormat_supportedLocalesOf_name, locales, options);
-        }), intlStaticMethodThisArg);
+    const dateTimeFormat_supportedLocalesOf_func = tagPublicFunction(dateTimeFormat_supportedLocalesOf_name, function (locales, options) {
+        return supportedLocalesOf_unconstructable(this, dateTimeFormat_supportedLocalesOf_name, undefined, locales, options);
+    });
+    const dateTimeFormat_supportedLocalesOf = _.bind(dateTimeFormat_supportedLocalesOf_func, intlStaticMethodThisArg);
 
     const getCanonicalLocales_name = "Intl.getCanonicalLocales";
-    const getCanonicalLocales = callInstanceFunc(FunctionInstanceBind, tagPublicFunction(getCanonicalLocales_name,
-        function getCanonicalLocales_dummyName(locales) {
-            return canonicalizeLocaleListWrapper(this, getCanonicalLocales_name, locales);
-        }), intlStaticMethodThisArg);
+    const getCanonicalLocales_func = tagPublicFunction(getCanonicalLocales_name, function (locales) {
+        return getCanonicalLocales_unconstructable(this, getCanonicalLocales_name, locales);
+    });
+    const getCanonicalLocales = _.bind(getCanonicalLocales_func, intlStaticMethodThisArg);
 
     // TODO: Bound functions get the "bound" prefix by default, so we need to remove it.
     // When https://github.com/Microsoft/ChakraCore/issues/637 is fixed and we have a way
     // to make built-in functions non-constructible, we can remove the call to
     // Function.prototype.bind (i.e. FunctionInstanceBind) and just rely on tagging instead of setting the "name" manually.
-    ObjectDefineProperty(collator_supportedLocalesOf, 'name', { value: 'supportedLocalesOf' });
-    ObjectDefineProperty(numberFormat_supportedLocalesOf, 'name', { value: 'supportedLocalesOf' });
-    ObjectDefineProperty(dateTimeFormat_supportedLocalesOf, 'name', { value: 'supportedLocalesOf' });
-    ObjectDefineProperty(getCanonicalLocales, 'name', { value: 'getCanonicalLocales' });
+    _.defineProperty(collator_supportedLocalesOf, 'name', { value: 'supportedLocalesOf' });
```

Comment:
Nit: Is it spec compliant to make the function name non-configurable?

Reply:
Will fix in the next PR